examples/AspNetCore/Controllers/WeatherForecastController.cs

```diff
@@ -24,16 +26,22 @@ public class WeatherForecastController : ControllerBase
 {
     private static readonly string[] Summaries = new[]
     {
-            "Freezing", "Bracing", "Chilly", "Cool", "Mild", "Warm", "Balmy", "Hot", "Sweltering", "Scorching",
+        "Freezing", "Bracing", "Chilly", "Cool", "Mild", "Warm", "Balmy", "Hot", "Sweltering", "Scorching",
     };
 
     private static readonly HttpClient HttpClient = new();
+    private static readonly ActivitySource MyActivitySource = new ActivitySource("Examples.AspNetCore");
+    private static readonly Meter MyMeter = new Meter("Examples.AspNetCore");
```

Comment:
Why is anything static? I think we need a DI centric approach to counters here. This doesn't look like something we'd tell people to do.

Reply:
Related threads:

* https://github.com/open-telemetry/opentelemetry-dotnet/pull/4133#discussion_r1093596387
* https://github.com/open-telemetry/opentelemetry-dotnet/pull/4133#discussion_r1093858644

Reply:
Injecting the activity source or meter is bad but injecting a service that has the instance is good (that's what we should do here).

Reply:
@davidfowl looks like @noahfalk captured it here https://github.com/open-telemetry/opentelemetry-dotnet/pull/4133#discussion_r1093818794.

Reply:
@CodeBlanch also hinted at this in his original comment. I am curious what your thoughts are on the where the responsibility lies for creating the actual `Instrument` (in this case a `Counter`)

Reply:
Short answer - If the Meter is shared via a DI container you probably want the Instruments also shared via the DI container. Most likely this means the code that created the Meter is creating the instruments.

Long answer - Meters hold a reference to all Instruments they create, so once created Instruments live as long as their parent Meter does. Assuming the Meter object has a long lifetime that will be reused for an unbounded number of requests, creating new instances of Instrument objects per-request would leak memory so we don't want to do that. The alternatives that don't leak require all code using the Instrument objects to be sharing the same bounded set of them. This means either that the instruments are singletons in the container or they are stored in fields of some other DI singleton, or some other mechanism is employed to share them such as a static cache.

More broadly folks who have looked into using Meters in DI have found them [lacking](https://github.com/dotnet/runtime/issues/77514). I would argue none of the requests I have seen represent an inability for Meters to be used with a DI container, but people's expectations change and they expect more functionality or alternate patterns in addition to just putting the objects in DI. I think the best pattern you can do today is to put the Meter + all the Instruments in a container type and let DI inject it. As we look into some of the requests folks are making I expect some new APIs will show up and best practice will shift.

Reply:
I would recommend declaring singleton service that holds meter and activity source instances and flowing that to sample code.

Reply:
@noahfalk @davidfowl thank you for the feedback. I pushed an update that I hope aligns with your desires. I named the type `TelemetryService` but maybe it should be `TelemetryContainer`, `TelemetryWrapper`, etc.